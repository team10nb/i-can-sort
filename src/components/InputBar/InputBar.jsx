import React from "react";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import {
    makeStyles,
    withStyles,
} from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
    root: {
        width: "780px",
        display: "flex",
        justifyContent: "center",
        alignContent: "flex-end",
        height: 40,
    },
    text: {
        marginRight:"10px",
    },
    button: {
        backgroundColor: "#52af77",
        color: "white",
        marginTop: "0px",
        borderRadius: 10,
        "&:hover, &$focusVisible": { backgroundColor: "#52af77"}
    },
}));

const CssTextField = withStyles({
    root: {
        "& label.Mui-focused": {
            color: "grey",
        },
        "& .MuiInput-underline:after": {
            borderBottomColor: "green",
        },
        "& .MuiOutlinedInput-root": {
            "& fieldset": {
                borderColor: "grey",
                borderRadius: 12,
            },
            "&:hover fieldset": {
                borderColor: "grey",
            },
            "&.Mui-focused fieldset": {
                borderColor: "green",
            },
        },
        width: "500px",
    },
})(TextField);

export default function InputBar(props) {
    const {
        defaultArr,
        isValid,
        wrongMsg,
        storeInput,
        checkFormat,
    } = props;

    const classes = useStyles();

    const error = isValid ? "" : "error";
    const label = isValid ? "Enter your own array" : "Invalid Input";
    const helper = isValid ? " " : wrongMsg;

    return (
        <div className={classes.root}>
            <div style={{width: "550px", display:"flex"}}>
            <CssTextField
                error={error}
                label={label}
                type='search'
                helperText={helper}
                variant='outlined'
                onChange={storeInput}
                onFocus={storeInput}
                defaultValue={defaultArr}
                className={classes.text}
                size='small'
            />
                <Button
                    variant='contained'
                    disableElevation
                    onClick={checkFormat}
                    className={classes.button}
                >
                    Create
                </Button>

            </div>
        </div>
    );
}