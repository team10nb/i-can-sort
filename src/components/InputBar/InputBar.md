### **Developer**

Ruizi Han, Shiliang Chen

Reference: MATERIAL-UI (https://material-ui.com)

### **Introduction**

A input bar with a button for user to define the array of numbers.

### **Import**

```html
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import {
    makeStyles,
    withStyles,
} from "@material-ui/core/styles";
```

### **Props**

| Name         | Type   | Default | Description                                        |
| ------------ | ------ | ------- | -------------------------------------------------- |
| storeInput | fn     |         | To store the input into an array.         |
| checkFormat | fn     |         | A function to check whether the input is correct.         |
| wrongMsg         | string | 0       | Error message. |
| defaultArr         | string | 0       | Indicate the default set array of numbers in the form of string. |
| isValid          | boolean | 0       | Whether the content is valid by definition.               |

### **Demo**

```jsx
<InputBar
    defaultArr={[1,2,3,4,5,6,7]}
    isValid={false}
    wrongMsg={"Error message"}
/>
```
