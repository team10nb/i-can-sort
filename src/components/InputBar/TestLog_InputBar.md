#### ComponentName: InputBar

|Test No.|Date|Pass/Fail|Failed test(s)|Reason/Description|Modified version|
|-------|----|---------|-------------------|----------------|----------------|
|1 | 2.14 | Fail | All | props error should have boolean value instead of string|version 1|

##### Code - version 1
```javascript
// const error = isValid ? "" : "error";
const error = isValid ? false : true;
```