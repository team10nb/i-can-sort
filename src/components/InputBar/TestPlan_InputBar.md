## Test Plan 
--------
<br>

#### ComponentName: InputBar

|Test Name|Description|Operation|Expected Outcome|
|----|------|--------|----------------|
| should contain buttons || | Contain one TextField and two buttons |
| should call functions |When click buttons, corresponding functions should be called| mock click | functions are called once |
| should match Snapshot | Compare new snapshot with old one. |  | Snapshots are the same. |








