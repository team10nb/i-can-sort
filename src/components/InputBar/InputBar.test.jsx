/*
    Author: Yijie Lu, Ruizi Han
*/

import { Input } from "@material-ui/core";
import { render } from "@testing-library/react";
import InputBar from "./InputBar";
import userEvent from "@testing-library/user-event";

const mockHandleChange = jest.fn();
const mockCheckFormat = jest.fn();
const mockShuffle = jest.fn();

const props = {
    defaultArr: "2,1,6,10,9,12,8,3",
    isValid: true,
    wrongMsg: "Please enter no more than 15 numbers that between 1-30.",
    handleChange: mockHandleChange,
    checkFormat: mockCheckFormat,
    shuffle: mockShuffle,
    inputString: "inputString",
};

it("should contain buttons", () => {
    const { getAllByRole } = render(<InputBar {...props} />);
    const buttons = getAllByRole("button");
    expect(buttons.length).toBe(2);
});

it("should call functions", () => {
    const { getAllByRole } = render(<InputBar {...props} />);
    const buttons = getAllByRole("button");

    userEvent.click(buttons[0]);
    userEvent.click(buttons[1]);

    expect(mockCheckFormat).toBeCalledTimes(1);
    expect(mockShuffle).toBeCalledTimes(1);
});

it("should match snapshot", () => {
    const content = render(<InputBar {...props} />);
    expect(content).toMatchSnapshot();
});
