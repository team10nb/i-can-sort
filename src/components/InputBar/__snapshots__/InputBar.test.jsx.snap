// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-19"
      >
        <div
          style="width: 550px; display: flex;"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-23 makeStyles-text-20"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
            >
              Enter your own array using comma
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                type="search"
                value="inputString"
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-26 PrivateNotchedOutline-legendNotched-27"
                >
                  <span>
                    Enter your own array using comma
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled MuiFormHelperText-marginDense"
            >
              <span>
                ​
              </span>
            </p>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonCreate-21 MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Create
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonShuffle-22 MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Shuffle
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-19"
    >
      <div
        style="width: 550px; display: flex;"
      >
        <div
          class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-23 makeStyles-text-20"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
          >
            Enter your own array using comma
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              type="search"
              value="inputString"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-26 PrivateNotchedOutline-legendNotched-27"
              >
                <span>
                  Enter your own array using comma
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled MuiFormHelperText-marginDense"
          >
            <span>
              ​
            </span>
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonCreate-21 MuiButton-disableElevation"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Create
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonShuffle-22 MuiButton-disableElevation"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Shuffle
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
